### ISSUE TITLE
codefix convertToAsync does not add `await` for returned values

### ISSUE THREAD

**Issue by rradczewski** [Microsoft/TypeScript#27544]
First of all, thanks a lot for this refactoring, I think it's really useful and will push the adoption of async-await syntax :+1: 

This is a bug report related to [this Twitter conversation](https://twitter.com/rradczewski/status/1045754228218757122): The converted code does not `await` for `Promise.reject`, thus the runtime does not throw but returns `Promise.reject(3)`, which does not trigger the `catch`-clause but makes the function call return a rejected Promise.

I'm not 100% sure what the best conversion would be, but I assume it would be to always await for functions that could possibly return a `Promise` or expressions that are a `Promise`.

**TypeScript Version:**  Tested against [https://github.com/Microsoft/TypeScript/tree/v3.1.1](https://github.com/Microsoft/TypeScript/tree/v3.1.1) and 4ed85b7c44028970c7c3d3061ec2b1ab2bea3321

**Failing testcases:**
- On top of `master`: https://github.com/rradczewski/TypeScript/tree/bug-return-needs-await (https://github.com/Microsoft/TypeScript/compare/master...rradczewski:bug-return-needs-await)
- On top of `v3.1.1`: https://github.com/rradczewski/TypeScript/tree/bug-v3.1.1-return-needs-await (https://github.com/Microsoft/TypeScript/compare/v3.1.1...rradczewski:bug-v3.1.1-return-needs-await)

These assume that the correct behavior is to add `await` to a returned expression that yields a Promise.

**Search Terms:**
- async
- convertToAsync

**Code**

```ts
function testcase() {
  return Promise.resolve(1)
    .then(x => Promise.reject(2))
    .catch(err => console.log(err));
}
```
Running `testcase()` code will log `2` and the returned Promise will resolve to `undefined`.

**Expected behavior:**
```ts
async function testcase() {
    try {
        const x = await Promise.resolve(1);
        return await Promise.reject(2);
    }
    catch (err) {
        return console.log(err);
    }
}
```
A call to `testcase()` would print `2` and the returned Promise would resolve to `undefined` as above.

**Actual behavior:**
```ts
async function testcase() {
    try {
        const x = await Promise.resolve(1);
        return Promise.reject(2);
    }
    catch (err) {
        return console.log(err);
    }
}
```
A call to `testcase()` this won't print `2` but node will complain about an unhandled rejected promise.

**Related Issues:** 
- #25082 - The proposal, while mentioning "changed semantics", gives two conflicting examples:
  - *Catch handlers* has `await console.log(...)`
  - *Multiple .catch() calls* does not add `await` to `console.log`
- Possibly #27112 and #27156

**Comment by j-oliveras** [comment]
Tested with node 8.11.1 and it seems is necessary to await `Promise.reject` rejections but not resolve at return position.

**Comment by j-oliveras** [comment]
I found and explanation when is necessary an await: https://stackoverflow.com/a/42750371

With the await, it is catched by the catch block. Without it, returns the promise and then is resolved (or rejected in this case).

@uniqueiniquity maybe, `convertToAsync` should await the return if it inside a catch block.

**Comment by uniqueiniquity** [comment]
@rradczewski thanks for reporting, and @j-oliveras thanks for the investigation! I'll tackle this as soon as I can.
